integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/iterate/IterateJsonPathTest.java

```diff
@@ -0,0 +1,194 @@
+        Map<String, String> header = new HashMap<String, String>();
+        header.put("Content-Type", "application/json");
+
+        HttpResponse httpResponse = doPost(endpoint, input, header);
+        assertTrue(httpResponse.getData().contains("Alice"));
+        assertTrue(httpResponse.getData().contains("Bob"));
+        assertTrue(httpResponse.getData().contains("Camry"));
+    }
+
+    @Test(groups = "wso2.esb", description = "Testing Iterate mediator json support with Call mediator inside target")
+    public void testIterateMediatorFlowWithCallMediator() throws Exception {
+        URL endpoint = new URL(getApiInvocationURL("iteratesample7") + "/iteratejson7");
+        Map<String, String> header = new HashMap<String, String>();
+        header.put("Content-Type", "application/json");
+
+        HttpResponse httpResponse = doPost(endpoint, input, header);
+        assertTrue(httpResponse.getData().contains("Alice"));
+        assertTrue(httpResponse.getData().contains("Bob"));
+        assertTrue(httpResponse.getData().contains("Camry"));
+    }
+
+    @Test(groups = "wso2.esb", description = "Testing Iterate mediator json support with Sequential attribute")
+    public void testIterateMediatorFlowWithSequential() throws Exception {
+        URL endpoint = new URL(getApiInvocationURL("iteratesample8") + "/iteratejson8");
+        Map<String, String> header = new HashMap<String, String>();
+        header.put("Content-Type", "application/json");
+
+        HttpResponse httpResponse = doPost(endpoint, input, header);
+        assertTrue(httpResponse.getData().contains("Alice"));
+        assertTrue(httpResponse.getData().contains("Bob"));
+        assertTrue(httpResponse.getData().contains("Camry"));
+    }
+
+    private static HttpResponse doPost(URL endpoint, String postBody, Map<String, String> headers)
+            throws AutomationFrameworkException, IOException {
+        HttpURLConnection urlConnection = null;
+        try {
+            urlConnection = (HttpURLConnection) endpoint.openConnection();
+            try {
+                urlConnection.setRequestMethod("POST");
+            } catch (ProtocolException e) {
+                throw new AutomationFrameworkException("Shouldn't happen: HttpURLConnection doesn't support POST?? " +
+                        e.getMessage(), e);
+            }
+            urlConnection.setDoOutput(true);
+            urlConnection.setDoInput(true);
+            urlConnection.setUseCaches(false);
+            urlConnection.setAllowUserInteraction(false);
+            urlConnection.setReadTimeout(10000);
+            for (Map.Entry<String, String> e : headers.entrySet()) {
+                urlConnection.setRequestProperty(e.getKey(), e.getValue());
+            }
+            OutputStream out = urlConnection.getOutputStream();
+            try {
+                Writer writer = new OutputStreamWriter(out, "UTF-8");
+                writer.write(postBody);
+                writer.close();
```

Comment:
The stream should be close inside the finally block. Otherwise, if any error occurs in the previous lines, Stream will keep open. Please check this.